staff/miguel-angel-zapatero/precourse/airlines/index.js

```diff
@@ -0,0 +1,123 @@
+/* Programa una inferfaz de usuario para una aerolinea (por terminal...). Esta aerolinea dispondrá de 10 vuelos para el dia de hoy, para empezar, estos vuelos estarán declarados de manera global, cuando se llame a la función:
+
+Se preguntará por el nombre de usuario y dará la bienvenida.
+El usuario visualizará todos los vuelos disponibles de una forma amigable: El vuelo con origen: Barcelona, y destino: Madrid tiene un coste de XXXX€ y no realiza ninguna escala.
+A continuación, el usuario verá el coste medio de los vuelos.
+También podrá ver cuantos vuelos efectúan escalas.
+Sabiendo que los ultimos 5 vuelos (los últimos 5 ID's) son los últimos del día, muestra al usuario sus destinos. */
+
+let flights = [
+    { id: 00, to: 'Bilbao', from: 'Barcelona', cost: 1600, scale: false },
+    { id: 01, to: 'New York', from: 'Barcelona', cost: 700, scale: false },
+    { id: 02, to: 'Los Angeles', from: 'Madrid', cost: 1100, scale: true },
+    { id: 03, to: 'Paris', from: 'Barcelona', cost: 210, scale: false },
+    { id: 04, to: 'Roma', from: 'Barcelona', cost: 150, scale: false },
+    { id: 05, to: 'London', from: 'Madrid', cost: 200, scale: false },
+    { id: 06, to: 'Madrid', from: 'Barcelona', cost: 90, scale: false },
+    { id: 07, to: 'Tokyo', from: 'Madrid', cost: 1500, scale: true },
+    { id: 08, to: 'Shangai', from: 'Barcelona', cost: 800, scale: true },
+    { id: 09, to: 'Sydney', from: 'Barcelona', cost: 150, scale: true },
+    { id: 10, to: 'Tel-Aviv', from: 'Madrid', cost: 150, scale: false }
+];
+
+function showFlights(data) {
+    let allFlights = '';
+    // Podría cambiarlo para hacerlo mediante un forEach igual que en la función showScaleFlights.
+    for (let i = 0; i < data.length; i++) {
+        allFlights += `El vuelo con origen: ${data[i].from}, y destino: ${data[i].to} tiene un coste de ${data[i].cost}€`;
+        if (data[i].scale) {
+            allFlights += ' con escala.\n'
+        } else {
+            allFlights += ' y no realiza ninguna escala.\n'
+        }
+    }
+    return allFlights;
+}
+
+function showAvgCost(data) {
+    let sum = 0;
+    data.forEach(function(flight){
+        sum += flight.cost; 
+    }, sum);
+    return 'El coste medio de los vuelos es ' + (sum / data.length).toFixed(2) + '€';
```

Comment:
@maz-e Same here as above, this code is right, but you can store the operation into a variable and then print it in the return :)

Reply:
👍